### DISCUSSION TITLE
Номер карты пациента в МИС как Patient.Identifier

### DISCUSSION THREAD

**Discussion by eugenekogan-spb** [fhir-ru/core#128] [Административная РГ]
Использование номера карты в МИС в качестве идентификатора пациента является довольно типичной практикой, как в самой МИС, так и в регионе. Поэтому, видимо, следует на уровне Core определить, как это делать.
Есть вопросы.
1. Видимо, следует указывать Identifier.Type.Code = "PN" - Provider Number, не путать с номером медицинской карты.
2. Как формировать Identifier.System - верно ли, что все МИС следует регистрировать в едином репозитории?

**Comment by Vrodvi** [comment]
Provider Number имеет код "PRN", а не "PN".
К тому же "Provider" - это тот кто предоставляет медицинские услуги, т.е. это мед.персонал, см. [PRN](http://hl7.org/fhir/R4/v2/0203/index.html#v2-0203-PRN) и, следовательно, не может использоваться как тип идентификатора пациента (его номера карты).

Для номера карты, подходит "MR" - Medical record number.

По поводу Identifier.system - его просто нужно сделать обязательным для заполнения и указать, что пара Identifier.system\Identifier.value должна быть уникальна.

Дополнительно, использованные МИСами "пространства идентификаторов" (Identifier.system - _The namespace for the identifier value_) можно регистрировать в некотором репозитории (мы его даже начали вести - http://fhir-ru.zendoc.me/systems) в виде [namingsystem](https://www.hl7.org/fhir/namingsystem.html).

Но, внимание! 
Вопрос тут чуть более сложный чем просто регистрация МИС.
Поскольку одна МИС может быть установлена в разных учреждениях, где номера карт могут нумероваться одинаково, то пара идентификатор МИС+номер карты может оказаться не уникальной, что недопустимо.
Т.е. регистрировать нужно именно "пространства идентификаторов", которые используют МИСы.
Если некоторая МИС действительно использует неповторяющиеся номера карт во всех своих развёртываниях в разных учреждениях, то тогда хватит одной записи "пространства идентификаторов номеров карт" на одну МИС, но это вырожденный случай, а в общем случае их может быть больше одного.

В СЭМДах для этого определен специальный формат OID, который необходимо формировать на каждый экземпляр МИС:
> Идентификаторы экземпляров* МИС в медицинской организации - `OID_медицинской_организации.100.НомерМИС.НомерЭкзМИС`
> * Экземпляр МИС – это понятие, введенное для обеспечения уникальности пары корень+расширение в случае, когда ее нельзя
> обеспечить стандартными средствами имеющейся информационной системы.
> Например, МИС не обеспечивает уникальность записей, так как развернута в разных филиалах одного учреждения и в каждом
> из них работает на отдельной базе данных (без синхронизации между собой). В таком случае каждому из филиалов должен быть
> присвоен свой OID для экземпляров МИС («*.1», «*.2» и т.д.), и пара корень+расширение для каждой из записей будет
> гарантировано уникальной.
> В случае отсутствия необходимости в выделении нескольких экземпляров МИС данный узел в OID идентификаторе должен быть
> представлен константой (например, «*.1»).

Необходимо предложить альтернативу указанному в СЭМД подходу.

Но как мне кажется - формат "пространства идентификаторов" и их регистрация - должны быть только рекомендованы, но не требоваться. 
Единственное требование должно быть на уникальность пары Identifier.system\Identifier.value.

**Reply by eugenekogan-spb** [reply]
> Но как мне кажется - формат "пространства идентификаторов" и их регистрация - должны быть только рекомендованы, но не требоваться.
> Единственное требование должно быть на уникальность пары Identifier.system\Identifier.value.
> 

Здесь есть противоречие. Для того, чтобы обеспечить уникальность пары, необходимо обеспечить уникальность идентификатора МИС (поскольку различные МИС выдают пациентам идентификаторы не зная друг о друге).
А значит, либо надо предложить формат идентификатора МИС, либо централизованно вести их реестр.

Мне кажется, что придуманное для СЭМД решение вполне работоспособно и его можно оставить.

**Reply by Vrodvi** [reply]
Для этого существуют системы уникальной идентификации - OID или GUID.
Идентификаторы, присвоенные в соответствии с регламентами данных систем идентификации, гарантируют глобальную уникальность, так что эти идентификаторы можно централизовано не регистрировать.
Или регистрировать, но не делать это обязательным требованием.

**Reply by eugenekogan-spb** [reply]
Об этом и речь. Либо закрепляем формат идентификатора, либо ведем реестр. На мой взгляд, разумнее первое.

**Reply by eugenekogan-spb** [reply]
Про PRN: я ошибся, спасибо.
Но MR (номер медицинской карты), кажется, не подходит. 
В МИС обычно есть внутренний номер пациента, не обязательно равный номеру медицинской карты. В амбулаторной ситуации, когда у пациента карта одна, они могут совпадать. В ситуации стационара карта заводится на каждую госпитализацию, поэтому номер пациента в системе один, а номеров карт много.
Предлагаю рассмотреть PI - http://hl7.org/fhir/v2/0203/index.html#v2-0203-PI

**Reply by eugenekogan-spb** [reply]
Про OID и GUID: видимо, в нашем случае нет необходимости конструировать OID, поскольку организация, ответственная за номер пациента в МИС, будет указана в Identifier.Assigner.

**Reply by Vrodvi** [reply]
Согласен, для внутреннего номера пациента в МИС, PI похоже подходит.
- en: Patient internal identifier
- en: A number that is unique to a patient within an Assigning Authority. 
- http://hl7.org/fhir/v2/0203/index.html#v2-0203-PI

**Comment by Vrodvi** [comment]
Во время совещания 26.04.2022 обсудили фрагмент описания http://fhir-ru.zendoc.me/core.Patient#:patient-internal-identifier.
Выявлены некоторые проблемы с данным описанием, просьба рассмотреть:

# 1)
Описание желательно дополнить - необходима небольшая вводная о проблеме, например ваше описание подойдёт:

> В МИС обычно есть внутренний номер пациента, не обязательно равный номеру медицинской карты. В амбулаторной ситуации, когда у пациента карта одна, они могут совпадать. В ситуации стационара карта заводится на каждую госпитализацию, поэтому номер пациента в системе один, а номеров карт много.

# 2)
Identifier.Type = PI - это желательно указать со ссылкой на набор значений, из которого взят данный код (видимо http://terminology.hl7.org/ValueSet/v2-0203). 
Плюс этого кода нет в рекомендованном FHIR наборе значений http://hl7.org/fhir/ValueSet/identifier-type, значит в профиле нужно переопределить набор значений:

![image](https://user-images.githubusercontent.com/92928137/165523613-888c7a52-760e-4208-80f6-24604ebcd86a.png)

# 3) 
Взятый из СЭМД формат пространства идентификаторов OID_медицинской_организации.100.НомерМИС.НомерЭкзМИС исходит из предположения, что Мед.организация ведёт у себя список МИС и поэтому она может сказать какая МИС имеет какой номер.
Данный момент не обозначен в описании, а он важен, т.к. не ясно каким образом этот формат может использовать конкретная МИС. Нужно описать регламент получения МИС'ом пространства идентификаторов от мед.организации. 

# 4) 
В качестве продолжения проблемы "3)" имеет проблему с атрибутом `Identifier.Assigner`.
У вас написано:

> Identifier.Assigner – медицинская организация - владелец информационной системы, выдавшей идентификатор

Но проблема в том, что выдаёт идентификатор не медицинская организация, а именно информационная система. И указание в данном атрибуте мед.организации вместо МИС'а приведёт к неправильной интерпретации всего идентификатора.
Этот атрибут корректно использовать для общеизвестных реестров, которые действительно поддерживаются одной организацией, например для идентификатора СНИЛС можно указать что его assigner - Пенсионный фонд.

**Reply by Vrodvi** [reply]
@eugenekogan-spb 
Это по поводу http://fhir-ru.zendoc.me/meetings.2022-04-26:
> 1. В дискуссии 128 описать замечания и пожелания к разделу «Идентификатор пациента в МИС» страницы Пациент - Родионов

**Comment by eugenekogan-spb** [comment]
@Vrodvi Спасибо за комментарии.
2) Перенес коды идентификаторов в Гугл-таблицу. Добавил еще коды, которые можно рассмотреть для добавления в наш ValueSet. 
https://docs.google.com/spreadsheets/d/1t89yGe2I1RHikgSFsx21U5cjoBEF7GQ2zrdBRqYKVIU/edit?usp=sharing

Вопрос: как мы можем поменять ValueSet, когда он Extensible?

**Reply by Vrodvi** [reply]
> Вопрос: как мы можем поменять ValueSet, когда он Extensible?

Определить новый ValueSet, в котором указать, что он содержит полное наполнение исходного ValueSet'а плюс добавленные нами коды.

В спецификации есть аналогичный пример - http://hl7.org/fhir/valueset-example-yesnodontknow.json.html
![image](https://user-images.githubusercontent.com/92928137/167836164-119ba2ca-eeb6-4256-9001-6948ee377a5f.png)

В нашем случае:
```
{
  "resourceType": "ValueSet",
  "id": "<сформировать id по принятому регламенту>",
  ...
  "compose": {
    "include": [
      {
        "valueSet": [
          "http://hl7.org/fhir/ValueSet/identifier-type"
        ]
      },
      {
        "system": "http://terminology.hl7.org/ValueSet/v2-0203",
        "concept": [
          {
            "code": "PI",
            "display": "Patient internal identifier"
          },
          {
            "code": "BCT",
            "display": "Birth Certificate"
          },
          {
            "code": "и так далее",
            "display": "..."
          },
        ]
      }
    ]
  },
```

Далее, видимо, необходимо создать профиль на тип данных Identifier (IdentifierRu?) с заменённым ValueSet'ом и в профиле Patient требовать использование этого переопределённого типа данных.

**Reply by eugenekogan-spb** [reply]
Как тогда будет обеспечено соответствие нашего пациента базовому стандарту?

**Reply by Vrodvi** [reply]
Противоречия базовому стандарту нет, т.к. сила связи с данным набором значений - Extensible.

**Comment by eugenekogan-spb** [comment]
### 4)
@Vrodvi Не соглашусь. Верно, что технически идентификатор выдает экземпляр МИС (инсталляция МИС).
Но с юридической стороны за него отвечает организация, которая эксплуатирует эту инсталляцию МИС. 
Кроме того, в спецификации явно указано, что Assigner - это организация.
<img width="622" alt="image" src="https://user-images.githubusercontent.com/62525329/170243806-ff1a771b-f51a-4a2c-aad5-80ef84f92bec.png">
Применяя это решение, мы не нарушаем требования уникальности, поскольку уникальность должна быть обеспечена по паре System-Value, а не Assigner-Value.

**Reply by Vrodvi** [reply]
Да, assigner - это организация. В случае с идентификатором, выданным из МИС, организацией может быть организация-поставщик этой МИС.

Соглашусь, что если юридически идентификатор выдаётся мед. организацией, то её можно указывать в assigner, но тогда это желательно более явно обозначить в комментарии к данному атрибуту, т.к. исходное описание атрибута "_Organization that issued/manages the identifier._" даёт определённый простор для интерпретации.

**Reply by eugenekogan-spb** [reply]
Я интерпретирую так, что организация-поставщик МИС отвечает за программный продукт, ПО. ПО, условно, поставляется с пустой базой пациентов. Медицинская организация, эксплуатирующая МИС, является оператором информационной системы, то есть отвечает за все что происходит в БД, в частности, за регистрацию в ней карточек пациентов и присвоение им номеров.
Из этого я делаю вывод, что юридически за выдачу номеров отвечает медицинская организация.

В комментарии к атрибуту добавил оговорку, что это не поставщик МИС.

**Comment by eugenekogan-spb** [comment]
### 3)
@Vrodvi я уточнил описание - проверьте, пожалуйста: http://fhir-ru.zendoc.me/core.Patient#:patient-internal-identifier

**Reply by eugenekogan-spb** [reply]
и @OlegPenzin проверьте пожалуйста
